decoder/vaapiDecoderJPEG_unittest.cpp

```diff
@@ -139,24 +403,41 @@ VAAPIDECODER_JPEG_TEST(Factory)
     doFactoryTest(mimeTypes);
 }
 
-VAAPIDECODER_JPEG_TEST(Decode_Simple)
+template <std::size_t N>
+void decodeJpeg(const std::tr1::array<uint8_t, N>& data, uint32_t fourcc)
 {
     VaapiDecoderJPEG decoder;
     VideoConfigBuffer config;
     VideoDecodeBuffer buffer;
 
     config.flag = 0;
 
-    buffer.data = const_cast<uint8_t*>(g_SimpleJPEG.data());
-    buffer.size = g_SimpleJPEG.size();
+    buffer.data = const_cast<uint8_t*>(data.data());
+    buffer.size = data.size();
     buffer.timeStamp = 0;
 
     ASSERT_EQ(YAMI_SUCCESS, decoder.start(&config));
     ASSERT_EQ(YAMI_DECODE_FORMAT_CHANGE, decoder.decode(&buffer));
     ASSERT_EQ(YAMI_SUCCESS, decoder.decode(&buffer));
     ASSERT_EQ(YAMI_SUCCESS, decoder.decode(&buffer));
 
-    EXPECT_TRUE(decoder.getOutput());
+    SharedPtr<VideoFrame> output = decoder.getOutput();
+    ASSERT_TRUE(output);
+    EXPECT_EQ(fourcc, output->fourcc);
+}
+
+VAAPIDECODER_JPEG_TEST(Decode_Simple)
+{
+    decodeJpeg(g_SimpleJPEG, YAMI_FOURCC_444P);
+    decodeJpeg(g_JPEG420, YAMI_FOURCC_IMC3);
+    /* this will trigger bug in dirver,
+     * see https://bugs.freedesktop.org/show_bug.cgi?id=97633c for details
+     */
+    /*
+    decodeJpeg(g_JPEG422H, YAMI_FOURCC_422H);
+    decodeJpeg(g_JPEG422V, YAMI_FOURCC_422V);
+    decodeJpeg(g_JPEG444P, YAMI_FOURCC_444P);
```

Comment:
From $va/test/decode/tinyjpeg.c, It only support u,v  h_sampling_factor 1. It expect 444p's h_sampling_factor[] equals to 1, 1, 1. But ffmpeg's will set 444P's h_sampling_factor[] to 4,4,4.  From my view,ffmpeg is no problem, and it is wildly used, we should support this in driver.

g_SimpleJPEG  is 444P but it's h_sampling_factor[] is 1,1,1. So it's no problem.

```
#ifdef SANITY_CHECK
  if (   (priv->component_infos[cY].Hfactor < priv->component_infos[cCb].Hfactor)
      || (priv->component_infos[cY].Hfactor < priv->component_infos[cCr].Hfactor))
    error("Horizontal sampling factor for Y should be greater than horitontal sampling factor for Cb or Cr\n");
  if (   (priv->component_infos[cY].Vfactor < priv->component_infos[cCb].Vfactor)
      || (priv->component_infos[cY].Vfactor < priv->component_infos[cCr].Vfactor))
    error("Vertical sampling factor for Y should be greater than vertical sampling factor for Cb or Cr\n");
  if (   (priv->component_infos[cCb].Hfactor!=1) 
      || (priv->component_infos[cCr].Hfactor!=1)
      || (priv->component_infos[cCb].Vfactor!=1)
      || (priv->component_infos[cCr].Vfactor!=1))
    printf("ERROR:Sampling other than 1x1 for Cr and Cb is not supported");
#endif
```